apps/server/src/modules/video-conference/service/video-conference.service.spec.ts

```diff
@@ -1176,39 +1212,47 @@ describe(VideoConferenceService.name, () => {
 		});
 
 		describe('when conference scope is VideoConferenceScope.VIDEO_CONFERENCE_ELEMENT', () => {
-			it('should call boardNodeService.findByClassAndId', async () => {
-				const { user, userId, conferenceScope, scopeId } = setup(VideoConferenceScope.VIDEO_CONFERENCE_ELEMENT);
+			const setupForElement = () => {
+				const { user, userId, conferenceScope, element, roomUser, scopeId } = setup(
+					VideoConferenceScope.VIDEO_CONFERENCE_ELEMENT
+				);
 				userService.findById.mockResolvedValue(user);
+				boardNodeService.findByClassAndId.mockResolvedValue(element);
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(roomUser);
+
+				return { userId, scopeId, conferenceScope };
+			};
+
+			it('should call boardNodeService.findByClassAndId', async () => {
+				const { userId, conferenceScope, scopeId } = setupForElement();
 
 				await service.getUserRoleAndGuestStatusByUserIdForBbb(userId, scopeId, conferenceScope);
 
 				expect(boardNodeService.findByClassAndId).toHaveBeenCalledWith(VideoConferenceElement, scopeId);
 			});
 
 			it('should call userService.findById', async () => {
-				const { user, userId, conferenceScope, scopeId } = setup(VideoConferenceScope.VIDEO_CONFERENCE_ELEMENT);
-				userService.findById.mockResolvedValue(user);
+				const { userId, conferenceScope, scopeId } = setupForElement();
 
 				await service.getUserRoleAndGuestStatusByUserIdForBbb(userId, scopeId, conferenceScope);
 
 				expect(userService.findById).toHaveBeenCalledWith(userId);
 			});
 
 			it('should return the user role and guest status for a video conference element conference', async () => {
-				const { user, userId, conferenceScope, scopeId } = setup(VideoConferenceScope.VIDEO_CONFERENCE_ELEMENT);
-				courseService.findById.mockResolvedValue(courseFactory.buildWithId({ name: 'Course' }));
-				userService.findById.mockResolvedValue(user);
+				const { userId, conferenceScope, scopeId } = setupForElement();
 
 				const result = await service.getUserRoleAndGuestStatusByUserIdForBbb(userId, scopeId, conferenceScope);
 
-				expect(result).toEqual({ role: BBBRole.MODERATOR, isGuest: false });
+				expect(result).toEqual({ role: BBBRole.VIEWER, isGuest: false });
 			});
 		});
 
 		describe('when conference scope is VideoConferenceScope.EVENT', () => {
 			it('should throw a ForbiddenException if the user is not an expert for an event conference', async () => {
 				const { userId, scopeId, team } = setup(VideoConferenceScope.EVENT);
```

Comment:
just for consistency, we have setupForElement and so on, but not for team events?

Reply:
Yes right, I left that out because it's only used once.